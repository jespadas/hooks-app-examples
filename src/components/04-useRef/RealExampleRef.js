import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

import '../02-useEffect/effects.css';

export const RealExampleRef = () => {

    /*
        useRef(initialValue) is a built-in React hook that accepts one
        argument as the initial value and returns a reference (aka ref). 
        A reference is an object having a special property current.
    */

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>Example Ref</h1>
            <hr />

            {show && <MultipleCustomHooks />}

            <button
                className="btn btn-primary mt-3"
                onClick={() => {
                    setShow(!show);
                }}
            >
                Show/Hide
            </button>


        </div>
    )
}

