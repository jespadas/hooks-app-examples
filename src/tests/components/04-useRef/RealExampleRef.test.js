import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';


describe('Test in <RealExampleRef />', () => {

    const wrapper = shallow(<RealExampleRef />);

    test('Should be displayed correctly', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

    });

    test('Should show component <MultipleCustomHooks />', () => {

        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

    })


})
