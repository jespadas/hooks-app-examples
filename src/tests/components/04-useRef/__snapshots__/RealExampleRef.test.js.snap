// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RealExampleRef /> Should be displayed correctly 1`] = `
<div>
  <h1>
    Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
